axon-framework/upgrading-to-4-7.md

```diff
@@ -0,0 +1,119 @@
+| org.axonframework.common.jpa.SimpleEntityManagerProvider                          | org.axonframework.common.legacyjpa.SimpleEntityManagerProvider                          |
+| org.axonframework.eventhandling.deadletter.jpa.DeadLetterJpaConverter             | org.axonframework.eventhandling.deadletter.legacyjpa.DeadLetterJpaConverter             |
+| org.axonframework.eventhandling.deadletter.jpa.EventMessageDeadLetterJpaConverter | org.axonframework.eventhandling.deadletter.legacyjpa.EventMessageDeadLetterJpaConverter |
+| org.axonframework.eventhandling.deadletter.jpa.JpaDeadLetter                      | org.axonframework.eventhandling.deadletter.legacyjpa.JpaDeadLetter                      |
+| org.axonframework.eventhandling.deadletter.jpa.JpaSequencedDeadLetterQueue        | org.axonframework.eventhandling.deadletter.legacyjpa.JpaSequencedDeadLetterQueue        |
+| org.axonframework.eventhandling.tokenstore.jpa.JpaTokenStore                      | org.axonframework.eventhandling.tokenstore.legacyjpa.JpaTokenStore                      |
+| org.axonframework.modelling.command.GenericJpaRepository                          | org.axonframework.modelling.command.legacyjpa.GenericJpaRepository                      |
+| org.axonframework.modelling.saga.repository.jpa.JpaSagaStore                      | org.axonframework.modelling.saga.repository.legacyjpa.JpaSagaStore                      |
+| org.axonframework.springboot.autoconfig.JpaAutoConfiguration                      | org.axonframework.springboot.autoconfig.legacyjpa.JpaJavaxAutoConfiguration             |
+| org.axonframework.springboot.autoconfig.JpaEventStoreAutoConfiguration            | org.axonframework.springboot.autoconfig.legacyjpa.JpaJavaxEventStoreAutoConfiguration   |
+| org.axonframework.springboot.util.jpa.ContainerManagedEntityManagerProvider       | org.axonframework.springboot.util.legacyjpa.ContainerManagedEntityManagerProvider       |
+| org.axonframework.messaging.interceptors.BeanValidationInterceptor                | org.axonframework.messaging.interceptors.legacyvalidation.BeanValidationInterceptor     |
+| org.axonframework.messaging.interceptors.JSR303ViolationException                 | org.axonframework.messaging.interceptors.legacyvalidation.JSR303ViolationException      |
+
+### Step 2
+
+If you have customized the `TokenEntry` or `AbstractTokenEntry`, rebase your changes on the current `TokenEntry`.
+This is required since the `AbstractTokenEntry` has been deprecated in favor of the `TokenEntry`.
+
+### Step 3
+
+If you have customized the `SagaEntry` or `AbstractSagaEntry`, rebase your changes on the current `SagaEntry`.
+This is required since the `AbstractSagaEntry` has been deprecated in favor of the `SagaEntry`.
+
+## Steps to upgrade from `javax` to `jakarta`
+
+### Step 1
+
+Remove any `javax` mentions in your codebase or replace them with respective `jakarta` ones.
```

Comment:
Any is a bit too strong. It's mainly `javax.persistance` and `javax.validation` that should be changed. It's still fine to use `javax.annotation` ant `javax.cache` for example.

Reply:
we might also point to open rewrite for this step, https://docs.openrewrite.org/reference/recipes/java/spring/boot3/springboot2to3migration, although I had problems using it with the hotel demo.

Reply:
Good point! I've adjusted this section accordingly.